lib/src/constants.dart

```diff
@@ -3,6 +3,25 @@
 // BSD-style license that can be found in the LICENSE file.
 
 import 'package:analyzer/dart/constant/value.dart';
+import 'package:analyzer/dart/element/element.dart';
+
+/// Throws an exception if [root] or its super(s) does not contain [name].
+void _assertHasField(ClassElement root, String name) {
+  var element = root;
+  while (element != null) {
+    final field = element.getField(name);
+    if (field != null) {
```

Comment:
[nit] consider collapsing this to 1 line so it doesn't take more visual weight than it earns.